### ISSUE TITLE
How does Lightgbm predicts for new categories on the test set

### ISSUE THREAD

**Issue by matan-schles** [Microsoft/LightGBM#1936]
Hi,

I'm working on a problem in which I have new categories for some features on my test set, that are not in my training set.
When I'm trying to predict on those features, I'm still getting a result, and not an error as I would expect.
So my question is what does the prediction function do in such cases?
How does it return a prediction?

I'm adding a python code example, to make my question more clear.

import numpy as np
import pandas as pd
import lightgbm as lgb

np.random.seed(1)
X1 = np.random.choice(["aaaaaa","bbb","ccc","ddd"],size=15)
X2 = np.random.choice(["ahhhm","avocado","eggplant"],size=15)
X1[14] = "other"
X2[14] = "other"
X = pd.DataFrame({'X1': X1, 'X2': X2}).apply(lambda col: col.astype('category'))
X_train = X.head(14)
Y_train = np.random.choice([0,1],size=14)
X_test = X.tail(1)

lgb_dat_train = lgb.Dataset(X_train,label = Y_train,categorical_feature = X_train.columns.tolist())
params = {
    'objective': 'binary',
    'metric': 'binary_logloss',
    'learning_rate': 0.01,
    'min_data_per_group' : 1,
    'num_iterations' :1,
    'num_leaves' :4,
    'min_data_in_leaf' :1
}
gbm = lgb.train(params=params,train_set=lgb_dat_train)
gbm.predict(X_test)
 
output: array([0.71213804])

Thanks a lot,
Matan

**Comment by guolinke** [comment]
LightGBM only uses the seen categories for prediction. 
For example, the trained split point may is  `if is A or is B then left child`, then for an unseen category, it will be in the right child.
